File: navigation/src/main/java/com/example/navigation/stepsEngine/field/DeviceField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;
import com.example.navigation.stepsEngine.payment.device.Device;

public class DeviceField extends Field {

    @Override
    public Device get(FlowState flowState) {
        return flowState.getPos().getDevice();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/DeviceField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState
import com.example.navigation.stepsEngine.payment.device.Device
import kotlinx.android.parcel.Parcelize

@Parcelize
class DeviceField : Field() {
    override operator fun get(flowState: FlowState): Device? {
        return flowState.pos.device
    }

    override fun getId(): String {
        return FieldName.DEVICE
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}